### PR TITLE
Redesign parts of the weapon sheet ⚔📜

### PR DESCRIPTION
**Pull request by giant-teapot**
Make a few visual improvements for the weapon sheet. It is based on the design of the current character sheet.

### New weapon sheet features:
 * clickable marks to increase/decrease the weapon bonus (wear/repair)
 * the artifact die is available as a dropdown menu
 * adds 3 tabs in the weapon sheet: attributes, artifact (for artifacts only), supply (for regular items)

weapon-sheet-preview

### Checklist
- [x] add clickable marks to increase/decrease the weapon bonus
- [x] add a dropdown menu  to select the artifact die
- [x] add 3 tabs to switch between: attributes, artifact-specific fields, crafting
- [x] ~~add checkboxes for features (hook, parrying)~~ descoped, left out for another time

### Translation
This change adds new strings in the software. Here is the current language support:
| Language  | Status |
| --- | ------------- |
| cs  | ❌ |
| de | ❌ |
| en | ✔|
| fr | ✔|
| pt-br | ❌ |
| sv | ✔|

### CODE REVIEW

**Comment by jimorie** [comment]
Nice work! I like not having to use the keyboard to update frequently changed values.

My concerns are things you already touch upon yourself, basically that we should do more to make it consistent:

* It looks a bit off having the border just around the header of the sheet. You mention adding tabs, so I guess you have plans to border up the bottom part of the sheet also?

* It is inconsistent with all the other item sheets. Do you plan to update the other sheets in similar manner? (Armors in particular comes to mind as the type of item where I would appreciate having marks to track the bonus value, more so than for weapons.)

**Comment by giant-teapot** [comment]
> My concerns are things you already touch upon yourself, basically that we should do more to make it consistent:

Oh, I very much agree with that statement. I hesitated before sending out this PR because − despite being functional and arguably a bit better − it's clearly nowhere near done.
I decided to send it anyway for two reasons:

1. Advertise what I'm currently doing ; it would be a shame if someone else was doing the same kind of work.
2. Get feedback ; because taste and vision might come in line with that kind of stuff.

> It looks a bit off having the border just around the header of the sheet. You mention adding tabs, so I guess you have plans to border up the bottom part of the sheet also?

That is correct. It's currently in progress, but that's basically what is going to happen.


> It is inconsistent with all the other item sheets. Do you plan to update the other sheets in similar manner? (Armors in particular comes to mind […])

You are absolutely right: the armor sheet should probably be next.

I'm glad you like the changes @jimorie . :)
The main challenge with this PR is that I voluntarily gave it a loose scope, but it gives you options:

1. If you _really_ value coherence over delay, we could agree that we should not merge until all items sheet were worked on.
2. Or we could agree to merge changes screen by screen: this branch is for weapons, we'll do armor next.
3. Or if you really want to push changes to the main branch: we could merge the redesign bit by bit.

Quite frankly, I'm more inclined towards option 2, but ultimately, I'm not the one merging, so I'll follow your guidelines. :)
By default I'll be adding commits to this PR during the coming days. Thanks for your comment, and please keep sending your feedback and ideas, it helps!

**Comment by jimorie** [comment]
I think it's great that you show what you're working on, both for feedback and updating other developers that work on the project. :)

If you're already working on more stuff for the weapon sheet I think option 2 sounds good. Let's keep this PR open until you're happy with that.

**Comment by giant-teapot** [comment]
@jimorie I've marked the PR as draft and added a checklist of things I'll be working on to clarify the scope and remaining work.
Keep you posted.

**Comment by aMediocreDad** [comment]
Hey! Thank you @giant-teapot for contributing!

I see you plan to separate out tabs, and that's a great idea. One of the feature requests I have for the system (and which I plan to work on once I am done with the content modules) is a template restructuring that will inevitably touch upon your work here. Artifacts will be treated as a common template shared between items and a such will be a toggleable state allowing you to toggle whether armor, gear or a weapon is an artifact or not. This will remove unnecessary clutter and data away from items that are not classified as artifacts. 

This is going to be part of a larger refactoring of the code, so I am not certain when exactly when these changes will see implementation. That said, I wanted to give you an early heads up, but would still love to see your work implemented until the system sees an overhaul.

**Comment by jimorie** [comment]
@giant-teapot I can contribute with Swedish translations of your strings:

```
  "ARTIFACT.REGULAR": "Ingen (vanligt föremål)",
  "ARTIFACT.MIGHTY": "Mäktig",
  "ARTIFACT.EPIC": "Episk",
  "ARTIFACT.LEGENDARY": "Legendarisk",
```

**Comment by giant-teapot** [comment]
Still room for improvement, but it's a start.
Happy I could play with this, helped me find my footing. I have a clearer picture on how the thing works and the state of the code. For instance, I now get the point of the refactoring @aMediocreDad mentioned : it's true that there's some duplications/missed opportunities around items… (but that's a discussion for another time)

Anyway, let me know if you find something wrong or if you want some adjustments made.

**Comment by aMediocreDad** [comment]
> Still room for improvement, but it's a start.
> Happy I could play with this, helped me find my footing. I have a clearer picture on how the thing works and the state of the code. For instance, I now get the point of the refactoring @aMediocreDad mentioned : it's true that there's some duplications/missed opportunities around items… (but that's a discussion for another time)
> 
> Anyway, let me know if you find something wrong or if you want some adjustments made. Go play with it and let me know.

Looks great! Just played with it a bit and I love the changes. This was one of the things I wanted to see done so thank you for contributing, truly! 💜

**Review by aMediocreDad — APPROVED** [review]
Approved! Adds a much needed change:)

File: style/item.css
Comment (jimorie) [inline style/item.css:94]: Preferential: Use the same layout for the supply and the crafting fields. Having labels in front in of the boxes and above in the other looks a bit off.

**Review by jimorie — COMMENTED** [review]
I'll echo @aMediocreDad and say thank you so much for this! It feels very nice with the tabs, I definitely think that's the way to go in order to keep the sheet more manageable as the number of fields and stuff on there grows.

I have added some comments to the code which are mostly just my preferences, and you can consider them or dismiss them as you like. I think this is a solid contribution regardless!

The one comment about the textareas and handlebar partials we probably do want to address before merging, however.

File: model/weapon.html
Comment (jimorie) [inline model/weapon.html:30]: There are some CSS rules that rely on this class being in the path. (E.g. the roll-modifiers are missing some styling because of this.) The easiest way to avoid the styles breaking is probably to re-introduce this class somewhere in the reworked HTML. (I see you do it for the artifact tab, but not on the supply tab.)

File: lang/en.json
Comment (giant-teapot) [reply lang/en.json:90]: Well, what you have in this tab covers as much crafting as it is about purchasing.
I liked the sound of "supply" because − I think − it sounds generic enough to cover both. What do you think?

File: style/item.css
Comment (giant-teapot) [reply style/item.css:96]: You're right, I just got a bit sloppy in the end. :P
I'll polish that up a bit.

File: style/item.css
Comment (giant-teapot) [reply style/item.css:32]: Good catch as well.

**Comment by giant-teapot** [comment]
Okay quite a few things to address before we send it in orbit. I'll work on it tomorrow.
I think we can wrap this up by the end of the week.

File: model/weapon.html
Comment (giant-teapot) [reply model/weapon.html:43]: I agree 100%. Also agree to leave it alone for now. :)

File: model/weapon.html
Comment (giant-teapot) [reply model/weapon.html:59]: wat 😅
Thanks for pointing that out.

File: model/weapon/tab/weapon_attributes.html
Comment (giant-teapot) [reply model/weapon/tab/weapon_attributes.html]: Tried it but as I feared, with the width of the window, it makes the tab bar look cluttered. I'll keep your remark in mind for later though.

**Comment by giant-teapot** [comment]
Allright, it should be all better now. :)
Thanks again for your comments @jimorie 

@aMediocreDad If my micro-commits bug you I can squash some of them, I don't mind. Let me know.

**Comment by aMediocreDad** [comment]
Thank you for going through with updating and doing the fixes @jimorie suggested. I'll go through the changes tonight!

If it's not too much hassle it would be great if you could squash by feature so it's easier to reference later on.

File: lang/en.json
Comment (jimorie) [reply lang/en.json:89]: I think a name that isn't also the name of a field is better (unless the other fields on the tab are related to that field -- which I don't really think they are), but my two cents are as much worth as yours here. What about something like "ACQUISITION" then, if the tab is about how one would acquire this item? But you should definitely go with "SUPPLY" if that sounds best to you.

File: model/weapon/tab/weapon_attributes.html
Comment (jimorie) [reply model/weapon/tab/weapon_attributes.html]: Okay, thanks for trying it!

File: lang/en.json
Comment (giant-teapot) [reply lang/en.json:89]: Funny enough: I went for « _acquisition_ » in the french version (we don't really have a nice and direct equivalent for « supply ») with 4148defd
I'll make the change I think.

While we're at it, if you have something something swedish for both ~~`TAB.SUPPLY`~~ `TAB.ACQUISITION` and `TAB.ARTIFACT`, I would be much obliged. 😄🇸🇪

Comment (aMediocreDad) [reply lang/en.json:89]: In my opinion SUPPLY, PROVISION, or ACQUISITION. The thing about Supply is that it is already a term in game, and so to avoid conflict/confusion the other two work better. ACQUISITION is in my opinion rather unwieldy. The tab title should in my opinion be called "Craft & Supply" if it can fit.

**Review by aMediocreDad — APPROVED** [review]
Tested and fixes look good:)

**Comment by giant-teapot** [comment]
Alright, I think this time we're good to go!
@aMediocreDad I've only added one last thing: I've changed README.md to use the screenshot in this PR.

File: lang/en.json
Comment (jimorie) [reply lang/en.json:89]: Maybe remove the string if it isn't used?

File: lang/en.json
Comment (jimorie) [inline lang/en.json:91]: Cool! Here are Swedish translations:

```
  "TAB.CRAFT_SUPPLY": "TILLVERKNING",
  "TAB.ARTIFACT": "ARTEFAKT",
```

Or if your prefer a more direct translation of Craft & Supply:

```
  "TAB.CRAFT_SUPPLY": "HANTVERK & LAGER",
```

**Comment by giant-teapot** [comment]
🚀 Okay! Final version is here!
All translations are in, squashing complete and last minute checks performed by yours truly. Time to seal the deal!

**Comment by aMediocreDad** [comment]
Great work! Thanks for this:) I'll await @jimorie's review then commit and do a new release.

File: script/sheet/weapon.js
Comment (jimorie) [inline script/sheet/weapon.js:11]: Should probably be changed to `main`, right? Although this doesn't seem to actually matter.

**Review by jimorie — APPROVED** [review]
I checked it out too, looks good to me! :+1:

**Pull request merged**
